### DISCUSSION TITLE
How to use federation with a headless CMS?

### DISCUSSION THREAD

**Discussion by LALauw** [apollographql/apollo-server#7267] [Q&A]
I have a Nextjs server running with a graphql endpoint at "http://localhost:3000/api/graphql", but I'm also utilising a headless cms called Hygraph. The problem is that I want to use federation for both my own graphql api and the one from Hygraph. I have set it up as follows:

````typescript
const gateway = new ApolloGateway({
  supergraphSdl: new IntrospectAndCompose({
    subgraphs: [
      {
        name: "cms",
        url: "Hygraph API endpoint",
      },
      { name: "person", url: "http://localhost:3000/api/graphql" },
    ],
    introspectionHeaders: {
      Authorization: `Bearer MyToken`,
    },
  }),
});

const apolloServer = new ApolloServer({
  gateway,
});

export default startServerAndCreateNextHandler(apolloServer, {});

````

However I'm getting the following error:An error occurred during Apollo Server startup. All GraphQL requests will now fail. The startup error was: Couldn't load service definitions for 'cms'

What am I doing wrong here and has anybody knowledge how to use federation with a headless cms?

**Comment by trevor-scheer** [comment]
@LALauw we need some more information to be helpful here. A runnable reproduction we can run would be very helpful, with explicit instructions on how to clone / set up and trigger the error.

Do you at least have logs from Hygraph you can share? Some details as to why that request failed would be helpful.

**Comment by mtthsnc** [comment]
@LALauw did you ever figure this out? Dealing with the same issue.

**Reply by trevor-scheer** [reply]
I can be more helpful with a reproduction.

**Reply by mtthsnc** [reply]
All good. I ended up using a GraphQL Mesh as it satisfied my needs for now.
